multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -2771,10 +2771,19 @@ def export_onnx(
         if not onnx_path:
             onnx_path = default_onnx_path
 
+        device_type = "cuda" if torch.cuda.is_available() else "cpu"
+        device = torch.device(device_type)
+        self._model.to(device).eval()
+        batch = move_to_device(batch, device=device)
+
+        if type(self._model).__module__.endswith("timm_image"):
+            input_vec = tuple([batch[k] for k in self._model.input_keys])
+        else:
+            input_vec = batch
         torch.onnx.export(
             self._model.eval(),
-            batch,
```

Comment:
Can we still use the dict format as model input? It seems that @FANGAreNotGnu made it work when export hf_text models.

Reply:
The reason we are planning to remove away from dict input comes with two folds: 1) there are [limitations](https://pytorch.org/docs/stable/onnx.html#limitations) in exporting onnx, 2) there are lots of benefits if we keep model inputs / outputs clear, see https://github.com/huggingface/transformers/issues/16059

Reply:
The dict input design is to flexibly support various modalities, backbones, and fusions in the AutoMM. I think changing the design may cause issues for some cases. For example, it seems difficult to list arguments of a fusion model since it may fuse any combinations of image, text, and tabular backbones.

Reply:
For most functions in AutoMM, we have used the type hints. The dict design can make each model more modular. A batch may contain inputs for several backbones and a model only needs to extract necessary information from it. This is especially useful in various fusion settings.

Reply:
If the dict does not work, does it make sense for us to switch to dataclasses or namedtuple @liangfu @zhiqiangdon ?

Reply:
I think using namedtuple is feasible. Similar to https://github.com/huggingface/transformers/blob/45da7cec5aa1b1bf1031af9caa9085e95e262e11/examples/research_projects/bertabs/run_summarization.py#L28

Reply:
We can try namedtuple, but not sure whether it is compatible with onnx.

Reply:
namedtuple is now experimentally working as inputs for tracing, but not yet ready to be part of `collate` function. will try integrate it into `collate` function in a follow-up PR.